DBZ-668 Change the key schema for the heartbeat messages from String to Struct

omarsmak opened the pull request:
As @gunnarmorling recommended [here](https://issues.jboss.org/browse/DBZ-668?focusedCommentId=13548206&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13548206), I have changed the heartbeat key message to `Struct` from `String` in order to leverage the `ByLogicalTableRouter` SMT and route all heartbeat messages into a single topic instead of individual topics for each source task

gunnarmorling commented on debezium-core/src/main/java/io/debezium/heartbeat/HeartbeatImpl.java:
```diff
@@ -42,7 +46,11 @@
     static final String DEFAULT_HEARTBEAT_TOPICS_PREFIX = "__debezium-heartbeat";
 
-    private static Schema KEY_SCHEMA = Schema.STRING_SCHEMA;
+    private static final String SERVER_NAME_KEY = "name";
```
Let's name it "serverName", to make it a bit more contextual. This matches the name of the corresponding connector option (`database.server.name`).

omarsmak commented:
Thank you @gunnarmorling for the fast feedback. I have added the unit test and changed the key field name per your suggestion :). Please let me know if there is anything else I should take care of

gunnarmorling commented:
@omarsmak LGTM, thanks! Can you just rebase it to the latest master and get rid of the merge commit? In general, we prefer rebases over merges for the sake of a cleaner history. Once that's done, I'll merge it.

Thanks again, nice improvement!

omarsmak commented:
Rebased onto the latest master and removed the unwanted merge commit, thank you again @gunnarmorling !

gunnarmorling commented:
Excellent. Thanks a lot, @omarsmak!

gunnarmorling commented:
Merged.
